From 903c853b0a4b1d56d61d5661171102f5f369c27d Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:06 +0100
Subject: [PATCH] tracking-2.6.25-input_dev-cdev-union-removed.patch

struct input_dev in include/linux/input.h used to have a union
cdev which contained the associated device struct pointer.  This
got simplified out in 2.6.25, so this patch removes cdev from
our drivers that used it before.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/i2c/chips/pcf50606.c        |    1 -
 drivers/i2c/chips/pcf50633.c        |    1 -
 drivers/input/keyboard/neo1973kbd.c |    1 -
 drivers/input/keyboard/qt2410kbd.c  |    1 -
 4 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/i2c/chips/pcf50606.c b/drivers/i2c/chips/pcf50606.c
index aa841e3..2466bf8 100644
--- a/drivers/i2c/chips/pcf50606.c
+++ b/drivers/i2c/chips/pcf50606.c
@@ -1708,7 +1708,6 @@ static int pcf50606_detect(struct i2c_adapter *adapter, int address, int kind)
 	data->input_dev->name = "FIC Neo1973 PMU events";
 	data->input_dev->phys = "I2C";
 	data->input_dev->id.bustype = BUS_I2C;
-	data->input_dev->cdev.dev = &new_client->dev;
 
 	data->input_dev->evbit[0] = BIT(EV_KEY) | BIT(EV_PWR);
 	set_bit(KEY_POWER, data->input_dev->keybit);
diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 09d5cfd..cb4cad8 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -1732,7 +1732,6 @@ static int pcf50633_detect(struct i2c_adapter *adapter, int address, int kind)
 	data->input_dev->name = "GTA02 PMU events";
 	data->input_dev->phys = "FIXME";
 	data->input_dev->id.bustype = BUS_I2C;
-	data->input_dev->cdev.dev = &new_client->dev;
 
 	data->input_dev->evbit[0] = BIT(EV_KEY) | BIT(EV_PWR);
 	set_bit(KEY_POWER, data->input_dev->keybit);
diff --git a/drivers/input/keyboard/neo1973kbd.c b/drivers/input/keyboard/neo1973kbd.c
index 8b9a420..6fc3ac1 100644
--- a/drivers/input/keyboard/neo1973kbd.c
+++ b/drivers/input/keyboard/neo1973kbd.c
@@ -199,7 +199,6 @@ static int neo1973kbd_probe(struct platform_device *pdev)
 	input_dev->id.vendor = 0x0001;
 	input_dev->id.product = 0x0001;
 	input_dev->id.version = 0x0100;
-	input_dev->cdev.dev = &pdev->dev;
 	input_dev->private = neo1973kbd;
 
 	input_dev->evbit[0] = BIT(EV_KEY) | BIT(EV_SW);
diff --git a/drivers/input/keyboard/qt2410kbd.c b/drivers/input/keyboard/qt2410kbd.c
index fa4bf32..cd8d7c6 100644
--- a/drivers/input/keyboard/qt2410kbd.c
+++ b/drivers/input/keyboard/qt2410kbd.c
@@ -124,7 +124,6 @@ static int gta01kbd_probe(struct platform_device *pdev)
 	input_dev->id.vendor = 0x0001;
 	input_dev->id.product = 0x0001;
 	input_dev->id.version = 0x0100;
-	input_dev->cdev.dev = &pdev->dev;
 	input_dev->private = gta01kbd;
 
 	input_dev->evbit[0] = BIT(EV_KEY);
-- 
1.5.6.3

